wallet/src/multisig_account.rs

```diff
@@ -0,0 +1,232 @@
+        Self {
+            address: Address::from(compute_root_from_content_slice::<Blake2bHasher, PublicKey>(
+                public_keys,
+            )),
+            key_pair: key_pair.clone(),
+            min_signatures,
+            public_keys: public_keys.to_vec(),
+        }
+    }
+
+    /// Generates a new commitment pair containing a commitment and a random secret.
+    #[inline]
+    pub fn create_commitment(&self) -> CommitmentPair {
+        CommitmentPair::generate(&mut SecureRng::default())
+    }
+
+    /// Creates an unsigned transaction.
+    pub fn create_transaction(
+        &self,
+        recipient: Address,
+        value: Coin,
+        fee: Coin,
+        validity_start_height: u32,
+        network_id: NetworkId,
+    ) -> Transaction {
+        Transaction::new_basic(
+            self.address.clone(),
+            recipient,
+            value,
+            fee,
+            validity_start_height,
+            network_id,
+        )
+    }
+
+    /// Utility method that delinearizes and aggregates the provided slice of public keys.
+    pub fn aggregate_public_keys(public_keys: &[PublicKey]) -> PublicKey {
+        let mut sorted_public_keys = Vec::with_capacity(public_keys.len());
+        public_keys.clone_into(&mut sorted_public_keys);
+        sorted_public_keys.sort();
+
+        let public_keys_hash = hash_public_keys(&sorted_public_keys);
+        let delinearized_pk_sum: EdwardsPoint = public_keys
+            .iter()
+            .map(|public_key| public_key.delinearize(&public_keys_hash))
+            .sum();
+
+        let mut public_key_bytes: [u8; PublicKey::SIZE] = [0u8; PublicKey::SIZE];
+        public_key_bytes.copy_from_slice(delinearized_pk_sum.compress().as_bytes());
+        PublicKey::from(public_key_bytes)
+    }
+
+    /// Creates a partial signature of the provided transaction.
+    pub fn partially_sign_transaction(
+        &self,
+        transaction: &Transaction,
+        public_keys: &[PublicKey],
+        commitments: &[Commitment],
+        secret: &RandomSecret,
+    ) -> PartialSignature {
+        let mut sorted_public_keys = Vec::with_capacity(public_keys.len());
+        public_keys.clone_into(&mut sorted_public_keys);
+        sorted_public_keys.sort();
+
+        self.key_pair
+            .partial_sign(
+                &sorted_public_keys,
+                secret,
+                commitments,
+                transaction.serialize_content().as_slice(),
+            )
+            .0
+    }
+
+    /// Creates a signature proof. Panics if the length of `signatures` is not equal to `min_signatures`.
+    pub fn create_proof(
+        &self,
+        aggregated_public_key: &PublicKey,
+        aggregated_commitment: &Commitment,
+        signatures: &[PartialSignature],
+    ) -> SignatureProof {
+        assert!(
+            signatures.len() == self.min_signatures as usize,
+            "Number of signatures must be the same as the minimal signatures."
+        );
```

Comment:
How about returning `Option<SignatureProof>` instead of panicking? I think it's better to return an error or `None` in this case.